### ISSUE TITLE
orchestrator: plumb available availability zones

### ISSUE THREAD

**Issue by benesch** [MaterializeInc/materialize#11809]
This issue tracks wiring up the [availability zone option](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/20220413_cluster_replica.md#orchestrator-changes) for cluster replicas.

The following work items are in scope:

  * [x] Adding the `--availability-zone` command-line flag
  * [x] Validating the availability zone against the allowed availability zone in the coordinator
  * [x] Attaching the correct label to services with a constrained availability zone in the orchestrator

Adjusting the cloud environment controller to pass the `--availability-zone` argument in out of scope of this issue.

Blocked on #11808.

Part of epic #11240.

**Comment by teskje** [comment]
Validating AZs has been implemented as part of #12313, so there is nothing left to do here.
